lib/modules/cardiovascular_disease.rb

```diff
@@ -366,16 +407,72 @@ def self.perform_encounter(entity, time)
             patient.condition(diagnosis, time, :condition, :condition)
           end
         end
+
+        if entity[:careplan] && entity[:careplan][:cardiovascular_disease]
+          if !entity.record_synthea.careplan_active?(:cardiovascular_disease)
+            entity.record_synthea.careplan_start(:cardiovascular_disease, entity[:careplan][:cardiovascular_disease]['activities'], time, entity[:careplan][:cardiovascular_disease]['reasons'])
+          else 
+            entity.record_synthea.update_careplan_reasons(:cardiovascular_disease, entity[:careplan][:cardiovascular_disease]['reasons'],time)
+          end
+        elsif entity.record_synthea.careplan_active?(:cardiovascular_disease)
+          entity.record_synthea.careplan_stop(:cardiovascular_disease,time)
+        end
+
+        if entity[:medications]
+          [:clopidogrel, :simvastatin, :amlodipine, :nitroglycerin, :atorvasatatin, :captopril, :warfarin, 
+            :verapamil, :digoxin, :epinephrine, :amiodarone, :atropine].each do |med|
+            if entity[:medications][med]
+              # Add a prescription to the record if it hasn't been recorded yet
+              if !entity.record_synthea.medication_active?(med)
+                entity.record_synthea.medication_start(med, time, entity[:medications][med][1])
+              else
+                entity.record_synthea.update_med_reasons(med, entity[:medications][med][1], time)
+              end
+            elsif entity.record_synthea.medication_active?(med)
+              # This prescription can be stopped...
+              entity.record_synthea.medication_stop(med, time, :cardiovascular_improved)
+            end
+          end
+        end
+
+        if entity[:cardiovascular_procedures]
+          entity[:cardiovascular_procedures].each do |reason, procedures|
+            reason_code = COND_LOOKUP[reason][:codes]['SNOMED-CT'][0]
+            procedures.each do |proc|
+              if !entity.record_synthea.present[proc]
+                entity.record_synthea.procedure(proc, time, reason_code, :procedure, :procedure) 
+              end
+            end
+          end
+        end
       end
 
```

Comment:
The above 5 lines are one of the (few) things I dislike about ruby.  End end end end end end end end end end end...

(Nothing to do here.  I just like complaining.)